src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -189,11 +189,38 @@ private function flatten(array $array, array &$result, $keySeparator, $parentKey
             if (\is_array($value)) {
                 $this->flatten($value, $result, $keySeparator, $parentKey.$key.$keySeparator);
             } else {
-                $result[$parentKey.$key] = $value;
+                $result[$parentKey.$key] = $this->cleanValueByType($value);
             }
         }
     }
 
+    /**
+     * Ensures an actual value is used instead of a blank value when dealing
+     * with true, false and null (like a nullable bool in a database).
+     *
+     * @param string $value
+     *
+     * @return string|int
+     */
+    private function cleanValueByType($value)
+    {
+        if (false === $value) {
+            return 0;
+        }
+
+        if (null === $value) {
+            // fputcsv will enclose a space
+            // https://github.com/php/php-src/blob/master/ext/standard/file.c
+            return ' ';
```

Comment:
That's a behavior change. I'm not sure that this qualifies as a bug fix. I would not change that here.

Reply:
Thanks @fabpot, I've removed the null check now.

I'm not sure why Appveyor failed, but it looks like an unrelated issue.